### ISSUE TITLE
Question: where are all the settings in PageSettingsInterface actually set?

### ISSUE THREAD

**Issue by tremby** [verbb/formie#857]
I see in the GQL schema and document at https://verbb.io/craft-plugins/formie/docs/developers/graphql#the-pageSettingsInterface-interface that page -> settings has a lot of fields. But in the control panel, on the configuration panel for a given page, I only see page label and conditions. Where are all the other per-page settings I can get back via that field configured?

**Comment by engram-design** [comment]
Each pages' submit buttons are stored within the page settings, so that's where the bulk of settings come from. The docs are slightly out of date, compared to what's available in GraphiQL.

**Comment by tremby** [comment]
Right, I think you're telling me I click to edit the submit button at the bottom of the list of fields in order to affect these fields.

I hadn't found that dialog -- it wasn't intuitive to me that I could even edit the submit button. If I hover over the other fields I see a gear icon and then there's an "edit" option, but there's no such thing for the submit button.

Now that I see that the submit button can be edited and the dialog it gives, it certainly looks like it would affect the various fields I was seeing available on the page settings structure in GQL. However, I think there's a bug here -- I can't get any of them to save. I fill in values, such as for the button label, change the button position, add CSS classes, add container attributes and input attributes, click apply to close the modal. I see the new button text in the list of fields. I click save in the top right corner and immediately the button has reverted to its old text. If I click to edit it again, I still see the new text in the label box. But if I then refresh the page and click to edit the button again, I see the original text. The GQL output never updated.

The doc text for these fields isn't very helpful either. For example, `buttonsPosition` says "the page's button positions" and it's not clear what that means. Worse, `containerAttributes` says "the field's container attributes" but this is part of `PageSettingsInterface`, so it's not clear which field it means, or what container. `inputAttributes` being for "the field's input attributes" is more confusing still. Then the docs for `enablePageConditions` and `enableNextButtonConditions` both say "whether the page has conditions enabled", and `pageConditions` and `nextButtonConditions` both say "the page's conditions".

**Comment by engram-design** [comment]
Fair point, we'll add some UI to inform users of that. I'll look into that saving mishap, I've seen that only once before, but was unable to replicate. Refreshing the page and trying again should do the trick.

Sorry for the sloppy docs work, I'll get that fixed.

**Comment by tremby** [comment]
> Fair point, we'll add some UI to inform users of that.

:+1:

> Sorry for the sloppy docs work, I'll get that fixed.

:+1:

> I'll look into that saving mishap, I've seen that only once before, but was unable to replicate. Refreshing the page and trying again should do the trick.

Not on my side. I still see the same issue, even after refreshing, even on a fresh form.

https://user-images.githubusercontent.com/199635/165867658-4740c34c-f523-404c-9e32-227779d32f04.mp4

**Comment by engram-design** [comment]
Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.0-beta.2",
  "...": "..."
}
```

Then run `composer update`.

**Comment by tremby** [comment]
I think this isn't fully fixed.

The behaviour I'm seeing now is that it'll save these fields fine if it's the first time I tried since refreshing the page or doing some other full page reload. If I try to change them again, the changes don't apply.

In this video I demonstrate that.

https://user-images.githubusercontent.com/199635/165872977-dde4e37f-e4d3-4f34-915a-5f4143567d29.mp4

**Comment by tremby** [comment]
And I seem to be unable to save "next" button conditions or page conditions at all.

**Comment by tremby** [comment]
In fact, they do save, and I can get them out via GQL, but they don't repopulate themselves in the control panel dialogs.

**Comment by engram-design** [comment]
Aha, I believe that might've been the issue I was talking about, which is a Vue issue. Once saved, the page settings become un-reactive like you're seeing. Fixed in `craft-4`.

**Comment by tremby** [comment]
Mmm I feel like something might still be funky with these conditions settings, but I can't reproduce it all the time. I'll open a new ticket if I find a way; this one's dealt with. Thanks again!

**Comment by engram-design** [comment]
Will continue to investigate!
